### ISSUE TITLE
Problem with adding fregre.glm learner from fda. usc package

### ISSUE THREAD

**Issue by NvNorsinski** [mlr-org/mlr#2716]
Hello,
I am trying to add the learner fregre.glm from package fda.usc. For the integration I followed the code for the already implemented and very similar integration of classif.glm from the same package in RLearner_classif_fdausc.glm.R. But I am having troubles when it comes to using the implementation, because I get an error in the for the trainLearner function, which seems to have an error. In the following you can see the code for the integration. Maybe I am doing a beginner's :) mistake.
``` r
library(mlr)
library(fda.usc)
library(parallelMap)

makeRLearner.regr.fregre.glm = function() {
  makeRLearnerRegr(
    cl = "regr.fregre.glm",
    package = "fda.usc",
    par.set = makeParamSet(
      makeDiscreteLearnerParam(id = "family", default = "binomial()", values = list("binomial()", "gaussian()", "Gamma()", "inverse.gaussian()", "poisson()")),
      makeUntypedLearnerParam(id = "basis.x"),
      makeUntypedLearnerParam(id = "basis.b"),
      makeLogicalLearnerParam(id = "CV", default = FALSE)
    ),
    properties = c("functionals"),
    name = "Generalized Linear Models regression on FDA",
    short.name = "fregre.glm",
    note = "model$C[[1]] is set to quote(fregre.glm)"
  )
}

trainLearner.regr.fregre.glm = function(.learner, .task, .subset, .weights = NULL, ...) {

  # Get and transform functional data
  d = getTaskData(.task, subset = .subset, target.extra = TRUE, functionals.as = "matrix")
  fd = getFunctionalFeatures(d$data)
  # transform the data into fda.usc:fdata class type.
  data.fdclass = fda.usc::fdata(mdata = as.matrix(fd))
  # transform the data into fda.usc:fdata class type and save in a list
  dat = list(df = data.frame(d$target), x = data.fdclass)

  model = fda.usc::fregre.glm(d.target ~ x, data = dat)

  # Fix bug in package. The changed slot looks different when called with
  # `fda.usc::lassif.glm()` than just `classif.glm()`
  model$call[[1]] = quote(fregre.glm)

  return(model)
}

predictLearner.regr.fregre.glm = function(.learner, .model, .newdata, ...) {
  # transform the data into fda.usc:fdata class type.
  fd = getFunctionalFeatures(.newdata)
  nd = list(x = fda.usc::fdata(mdata = fd))

  predict(object = .model$learner.model, newx = nd, type = "response")

}



registerS3method("makeRLearner", "regr.fregre.glm",
                 makeRLearner.regr.fregre.glm)

registerS3method("trainLearner", "regr.fregre.glm",
                 trainLearner.regr.fregre.glm)

registerS3method("predictLearner", "regr.fregre.glm",
                 predictLearner.regr.fregre.glm)


parallelExport("trainLearner.regr.fregre.glm",
               "predictLearner.regr.fregre.glm")

```

<sup>Created on 2020-01-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by pat-s** [comment]
Thanks. Please open a PR with the learner code you want to add so we can better comment on single code snippets.

Also do not forget to add tests for the new learner.

cc @pfistfl

**Comment by pat-s** [comment]
Solved in #2718 but the classif counterpart is still missing.
